### ISSUE TITLE
Smart commit mode not working by default on connections made through CLI

### ISSUE THREAD

**Issue by wheinen** [dbeaver/dbeaver#21209]
### Description

When adding a new connection in DBeaver through the Command Line Interface, the connections never use Smart Commit, even when you have set your connection types to use Smart commit mode by default.

All my connection types are set to use 'Auto-commit mode by default' and 'Smart commit mode'.
![image](https://github.com/dbeaver/dbeaver/assets/137804655/4d3b4d65-5446-4601-87d7-33a47c5360eb)

The command I use to add a new connection:
dbeaver-cli.exe -con "driver=mysql|autoCommit=true|host=Hostname.host.com|port=3306|user=USER|database=DATABASE|password=PASSWORD|name=Dummy connection name"

After adding the connection and opening a script, the connection is not using smart commit mode:
![image](https://github.com/dbeaver/dbeaver/assets/137804655/a9a00da4-c6e4-4be3-aaeb-0f3ff5607248)

Note: This feature was working fine in an earlier version of DBeaver, before the 'smart commit mode' checkbox was moved from the 'Transactions' menu to the 'Connection Types' menu in the preferences.

### DBeaver Version

23.2.0

### Operating System

Windows 10

### Database and driver

MySQL 8
com.mysql.cj.jdbc.Driver
8.0.29

### Steps to reproduce

1. Set all your connection types to use Smart commit mode
2. Add a new connection through the CLI using the earlier command
3. Open a script with the new connection, and see that Smart Commit is not turned on

**Comment by fjongDEVENTit** [comment]
We are also experiencing this issue. +1

**Comment by Matvey16** [comment]
@wheinen Thank you for report.

Note: That is not necessaryly a cli issue. This is reproducible for any newly created connection. The default smart commit preference enables for new connections only after dbeaver restart

**Comment by wheinen** [comment]
@Matvey16 You are right! I thought it was the CLI, because when I added a new connection through the user interface it always used Smart Commit. However, I now found out the crucial step is clicking on the 'Connection details (name, type, ... )' button. 
If you do that, and then save the connection it will use smart commit. If you don't click it, it won't use smart commit.
![image](https://github.com/dbeaver/dbeaver/assets/137804655/e96a3356-7f86-47d5-aab1-c074b2a59c6d)
